hydrocron/db/track_ingest.py

```diff
@@ -0,0 +1,163 @@
+"""
+Hydrocron class to track status of ingested granules.
+"""
+
+# Standard Imports
+import datetime
+from datetime import timezone
+import logging
+
+# Third-party Imports
+import requests
+
+# Application Imports
+from hydrocron.api.data_access.db import DynamoDataRepository
+from hydrocron.utils import connection
+
+
+logging.getLogger().setLevel(logging.INFO)
+logging.basicConfig(format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
+                    datefmt='%Y-%m-%dT%H:%M:%S',
+                    level=logging.INFO)
+
+
+class Track:
+    """
+    Class to track the status of ingested granules and submit missed or
+    newly discovered granules for Hydrocron database ingestion.
+    """
+
+    CMR_API = "https://cmr.earthdata.nasa.gov/search/granules.umm_json"
+    PAGE_SIZE = 2000
+
+    def __init__(self, collection_shortname, collection_start_date, hydrocron_table):
+        self.collection_shortname = collection_shortname
+        self.cmr_granules = {}
+        self.hydrocron_granules = {}
+        self.hydrocron_table = hydrocron_table
+        self.revision_start = self._get_revision_start(collection_start_date)
+        self.revision_end = datetime.datetime.now(timezone.utc)  # TODO - Decide on latency and subtract from current datetime
```

Comment:
very sorry for the delay on this, for fwd stream the latency from observation to archival is 48 - 72 hours. In the case of a reprocessing campaign I don't think we'd want to calculate from observation time though, since that could be months/years. The nominal latency from time between CNM notification and successful ingest is ~3 min, which will be true for both forward and reprocessed granules

Reply:
Maybe we can set the latency to 1 hour and query from the calculated revision start date to the current time minus 1 hour? I will plan on this for now and we can always adjust - I actually might make the latency into an environment variable that we can adjust when needed. I will include this work as a part of #211.

And now I will merge this PR as I think all work is complete!